processor/configuration/src/main/java/com/redhat/service/smartevents/processor/AbstractGatewayValidator.java

```diff
@@ -0,0 +1,37 @@
+package com.redhat.service.smartevents.processor;
+
+import java.util.stream.Collectors;
+
+import com.networknt.schema.ValidationMessage;
+import com.redhat.service.smartevents.infra.models.gateways.Gateway;
+import com.redhat.service.smartevents.infra.validations.ValidationResult;
+
+public abstract class AbstractGatewayValidator<T extends Gateway> implements GatewayValidator<T> {
+
+    private ProcessorCatalogService processorCatalogService;
+
+    protected AbstractGatewayValidator() {
+    }
+
+    protected AbstractGatewayValidator(ProcessorCatalogService processorCatalogService) {
+        this.processorCatalogService = processorCatalogService;
+    }
+
+    protected ValidationResult applyAdditionalValidations(T gateway) {
+        return ValidationResult.valid();
+    }
+
+    @Override
+    public final ValidationResult isValid(T gateway) {
+        if (gateway.getParameters() == null) {
+            return ValidationResult.invalid();
+        }
+
+        com.networknt.schema.ValidationResult result = processorCatalogService.validate(getType(), gateway.getProcessorType(), gateway.getParameters());
+        if (!result.getValidationMessages().isEmpty()) {
```

Comment:
Note to @manstis: I'll need to refactor this to return multiple discrete results!